src/actors/resolver.rs

```diff
@@ -256,13 +252,20 @@ impl Handler<Connect> for Resolver {
                 msg.port.unwrap_or(0),
                 self.resolver.as_ref().unwrap(),
             )
-            .and_then(move |addrs, _, _| TcpConnector::with_timeout(addrs, timeout)),
+            .then(move |addrs, _, _| {
+                let res = match addrs {
+                    Ok(a) => Ok(TcpConnector::with_timeout(a, timeout)),
+                    Err(e) => Err(e),
+                };
+
+                res.unwrap()
```

Comment:
Pardon my interjection out of the blue, but doesn't this `unwrap` introduce a panic rather than piping the `ResolverError` out as error value?

Reply:
No problem at all! I agree it should go. Taken care of.